File: pset1/cash.c
#include <stdio.h>
#include <math.h>

int calc(int c);

int main(){
    float change = 0;
    char ch;
    printf("Change owed: ");
    while(scanf("%f", &change) != 1 || change <= 0){
        printf("Change owed: ");
        while( (ch=getchar())!='\n' && ch!=EOF );
    }
    
    int cents = round(change * 100);
    int coins = calc(cents);
    printf("Coins: %i\n", coins);
}

// enter cents return coins
int calc(int cents){
    int coins;
    // 25
    coins = cents / 25;
    cents -= coins * 25;
    
    // 10
    coins += cents / 10;
    cents -= (cents/10) * 10;
    
    // 5
    coins += cents / 5;
    cents -= (cents/5) * 5;
    
    // 1
    coins += cents / 1;
    cents -= (cents/1) * 1;

    return coins;
}

File: pset6/Cash/cash.py
change = float(0)
while (change <= 0):
    change = input("Change owned: ")
    try:
        change = float(change) * 100
    except ValueError:
        change = 0

coins = change / 25
coins = int(coins)
change -= coins * 25

coins += change / 10
coins = int(coins)
change -= int(change / 10) * 10

coins += change / 5
coins = int(coins)
change -= int(change / 5) * 5

coins += change / 1
coins = int(coins)
change -= int(change / 1) * 1

print(coins)
